### PR TITLE
Prevent AFK kick while watching end credits (#6239).

### LINKED ISSUE #6239: [Feature] Prevent AFK timeout when on end credits screen
### Is your feature request related to a problem?

Many servers use built in `player-idle-timeout` config option in server.properties. This option prevents AFKing too long on one spot and generally helps servers save resources which they can use elsewhere where people actually play - by kicking the player and unloading area around him. There is one special scenario where the player gets kicked out, despite game being designed in a way that there is nothing player can do to prevent it.
After defeating the dragon and jumping into end portal, end credits start to show, with special music and everything. Player can fully enjoy finishing the game, watching the credits, etc., feeling victorious, when all of a suddent he gets kicked for being idle for too long. End credits take total of [39 minutes, 30 seconds](https://minecraft.fandom.com/wiki/End_Poem) and I believe that when most servers set their player-idle-timeout, it's less than 40 minutes.

### Describe the solution you'd like.

I'd love to see an exception introduced which would prevent kicking players who are on end credit screen. Either built in or as a config option. Honestly, I have no idea where is player located physically in the world during end credit's scene, if he is visible standing in the end or is teleported into "nowhere", but I believe most servers can handle those few chunks in end being loaded during end credits - if it even loads some in the first place.

### Describe alternatives you've considered.

Disabling `player-idle-timeout` altogether.. Keeping all AFK players connected...

### Agreements

- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] I have ensured the feature I'm requesting isn't already in the latest supported Paper build.

### Other

Although this wouldn't be considered as a bug and it is probably present even in vanilla Minecraft without any plans to change it, I think that this "feature" fits into Paper's moto of fixing gameplay/mechanics inconsistencies. I also agree that most players just skip the end credit screen completely, but it's just like finishing movie in cinema and sitting through the credits... Minecraft has special song there which you can hear nowhere else and being kicked just breaks the immersion completely... Unlike for real AFK players which it was intended for..

### PR DESCRIPTION
**Pull request by NoahvdAa**
This PR implements #6239 by preventing an AFK kick if the player is switching dimensions and in the end.

### CODE REVIEW

File: patches/server/0731-Prevent-AFK-kick-while-watching-end-credits.patch [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+index 68c3d744717cef8eeb16d640877157340a45e81c..219e93937b7f8f6e8f77e8f1ba221409352df3fc 100644
+--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
++++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+@@ -151,6 +151,7 @@ import net.minecraft.world.level.block.entity.SignBlockEntity;
+ import net.minecraft.world.level.block.entity.StructureBlockEntity;
+ import net.minecraft.world.level.block.state.BlockBehaviour;
+ import net.minecraft.world.level.block.state.BlockState;
++import net.minecraft.world.level.dimension.DimensionType;
```
Comment (Machine-Maker) [inline patches/server/0731-Prevent-AFK-kick-while-watching-end-credits.patch]: Don't add an import, use the FQN on the change below

**Review by Machine-Maker — CHANGES_REQUESTED** [review]
I feel like there's a viewingCredits boolean on the player entity or something, would that work better than a dimension check?

**Comment by NoahvdAa** [comment]
> I feel like there's a viewingCredits boolean on the player entity or something, would that work better than a dimension check?

I tried looking for that as well, but couldn't find it.

**Comment by Machine-Maker** [comment]
I think `wonGame` is that boolean. Based on where its set, it looks like when its true, the player is viewing the credits? It gets set back to false when the player respawns

**Review by Machine-Maker — APPROVED** [review]
Needs testing to make sure `wonGame` is what i think it is.

**Comment by NoahvdAa** [comment]
> I think `wonGame` is that boolean. Based on where its set, it looks like when its true, the player is viewing the credits? It gets set back to false when the player respawns

That appears to be it, the only parts lines where `wonGame` is modified have to do with dimension switching. I've gone and updated the patch to use `wonGame`.

**Comment by NoahvdAa** [comment]
Whoops, I tried to rename the branch but that ended up closing the PR, sorry about that.

**Review by Proximyst — APPROVED** [review]

**Pull request merged**